### DISCUSSION TITLE
my recognition: tinybase let us not use react hook, right?

### DISCUSSION THREAD

**Discussion by roccho-dev** [tinyplex/tinybase#84] [Q&A] [answered]
my recognition: tinybase let us not use react hook, right?

**Comment by jamesgpearce** [comment]
Incorrect! You will need to import the optional `tinybase/ui-react` module which provides a huge collection of hooks for setting and listening to data. There's more [here](https://tinybase.org/guides/building-uis/using-react-hooks/).

eg:

```js
const store = createStore().setCell('pets', 'fido', 'color', 'brown');
const App = () => <span>{useCell('pets', 'fido', 'color', store)}</span>;

const app = document.createElement('div');
const root = ReactDOMClient.createRoot(app);
root.render(<App />);
console.log(app.innerHTML);
// -> '<span>brown</span>'

store.setCell('pets', 'fido', 'color', 'walnut');
console.log(app.innerHTML);
// -> '<span>walnut</span>'
```

Hope that helps!

**Comment by roccho-dev** [comment]
Thanks, @jamesgpearce 
Can I ask the independency of tinybase?

Q1.
I may have chosen my words poorly. I won't stick with vanillajs, but I do want to be as independent of React ( and any ui library ) as possible and consistently manage state in memory and even various types of storage. 
I think tinybase very useful for that, and I also recognize that tinybase provides a consistent abstraction from small state management to memory store to server store. 
Is this wrong?

https://riffle.systems/essays/prelude/

Q2.
I now recognize that `ui-react` is the new place to move/migrate from React state management. 
This mean we don't have to use React Hooks for managing state.
`ui-react` may also be for this case?
OR `ui-react` is same as react state management?

https://news.ycombinator.com/item?id=32871392
"Cool! Please take a look and see if it works out. I should point out that the ui-react library (that provides React bindings) does _not_ depend on ReactDOM, so you can use it just as easily in React Native."

Q3.
On the other hand, I wondered if it is possible to implement the same functionality as `ui-react` with just the core functionality that `ui-react` does not have? 


p.s. By the way, I said "let" in my poor English because of my wish to be "as independent from any ui library as possible"

**Reply by jamesgpearce** [reply, marked as answer]
Sorry I missed this follow up! You should be able to just use TinyBase on its own without adding the ui-react module, no?
